src/frontend/components/property-type/reference/edit.tsx

```diff
@@ -13,67 +13,78 @@ type SelectRecordEnhanced = SelectRecord & {
   record: RecordJSON;
 }
 
-class Edit extends React.Component<CombinedProps> {
-  constructor(props: CombinedProps) {
-    super(props)
-    this.loadOptions = this.loadOptions.bind(this)
-    this.handleChange = this.handleChange.bind(this)
-  }
+const Edit: FC<CombinedProps> = (props) => {
+  const { onChange, property, record, theme } = props
 
-  handleChange(selected: SelectRecordEnhanced): void {
-    const { onChange, property } = this.props
+  const handleChange = (selected: SelectRecordEnhanced): void => {
     if (selected) {
       onChange(property.name, selected.value, selected.record)
     } else {
       onChange(property.name, '')
     }
   }
 
-  async loadOptions(inputValue: string): Promise<Array<SelectRecordEnhanced>> {
-    const { property } = this.props
+  const loadOptions = async (inputValue: string): Promise<SelectRecordEnhanced[]> => {
     const api = new ApiClient()
 
     const records = await api.searchRecords({
       resourceId: property.reference as string,
       query: inputValue,
     })
-    return records.map((record: RecordJSON) => ({
-      value: record.id,
-      label: record.title,
-      record,
+    return records.map((r: RecordJSON) => ({
+      value: r.id,
+      label: r.title,
+      record: r,
     }))
   }
+  const error = record.errors && record.errors[property.name]
 
-  render(): ReactNode {
-    const { property, record, theme } = this.props
-    const error = record.errors && record.errors[property.name]
+  const selectedId = record?.params[property.name] as string | undefined
+  const [loadedRecord, setLoadedRecord] = useState<RecordJSON | undefined>()
+  const [loadingRecord, setLoadingRecord] = useState(0)
+  const selectedValue = record?.populated[property.name] ?? loadedRecord
+  const selectedOption = (selectedId && selectedValue) ? {
+    value: selectedValue.id,
+    label: selectedValue.title,
+  } : {
+    value: '',
+    label: '',
+  }
+  const styles = selectStyles(theme)
 
-    const reference = record.populated && record.populated[property.name]
-    const selectedOption = reference ? {
-      value: reference.id,
-      label: reference.title,
-    } : {
-      value: '',
-      label: '',
+  useEffect(() => {
+    if (!selectedValue && selectedId) {
+      setLoadingRecord(c => c + 1)
+      const api = new ApiClient()
+      api.recordAction({
+        actionName: 'show',
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        resourceId: property.reference!,
```

Comment:
I would throw error on top of the function if property doesn't have reference. Now it can be also used by anyone with `BasePropertyComponent.Reference.Edit` so property.reference can be empty in this case.

Reply:
That would only happen if AdminBro was misconfigured, right? I mean, it would need a user having access to a resource which references another resource that the user does not have access to, am I understanding this correctly?